packages/block-library/src/query/index.php

```diff
@@ -5,12 +5,105 @@
  * @package WordPress
  */
 
+/**
+ * Modifies the static `core/query` block on the server.
+ *
+ * @since X.X.X
+ *
+ * @param array  $attributes Block attributes.
+ * @param string $content    Block default content.
+ * @param string $block      Block instance.
+ *
+ * @return string Returns the modified output of the query block.
+ */
+function render_block_core_query( $attributes, $content, $block ) {
+	if ( $attributes['enhancedPagination'] ) {
+		$p = new WP_HTML_Tag_Processor( $content );
+		if ( $p->next_tag() ) {
+			// Add the necessary directives.
+			$p->set_attribute( 'data-wp-interactive', true );
+			$p->set_attribute( 'data-wp-navigation-id', 'query-' . $attributes['queryId'] );
+			$p->set_attribute(
+				'data-wp-context',
+				wp_json_encode( array( 'core' => array( 'query' => (object) array() ) ) )
+			);
+			$content = $p->get_updated_html();
+
+			// Mark the block as interactive.
+			$block->block_type->supports['interactivity'] = true;
+
+			// Add a div to announce messages using `aria-live`.
+			$last_div_position = strripos( $content, '</div>' );
+			$content           = substr_replace(
+				$content,
+				'<div
+					style="position:absolute;clip:rect(0,0,0,0);width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;border:0;"
+					aria-live="polite"
+					data-wp-text="context.core.query.message"
+				></div>
+				<div
+					class="wp-block-query__enhanced-pagination-animation"
+					data-wp-class--start-animation="selectors.core.query.startAnimation"
+					data-wp-class--finish-animation="selectors.core.query.finishAnimation"
+				></div>',
+				$last_div_position,
+				0
+			);
+
+			// Use state to send translated strings.
+			wp_store(
+				array(
+					'state' => array(
+						'core' => array(
+							'query' => array(
+								'loadingText' => __( 'Loading page, please wait.' ),
+								'loadedText'  => __( 'The new page has been loaded. You can continue navigating.' ),
```

Comment:
> "Page Loaded" is probably enough here.

Fixed in https://github.com/WordPress/gutenberg/pull/53812/commits/fce30323e66d2ebd2b0c25c298e10325840eea47.

> Also, I believe some kind of announcement of the dynamically loaded content is helpful.

I will do this in a follow-up PR. 👍

Reply:
Hi, @jeryj. I was about to open a PR to address these changes. 🙂

> Maybe wrapping the posts container in an aria-live?

I just tried so, but the screen reader announces lots of changes (the whole list, basically). Is there a way to summarize the changes somehow? 😅

Reply:
Yeah, I was probably wrong to have it be wrapped in `aria-live`. I know you can press `ctrl` to stop the announcements though if you don't want to hear all of them. Maybe it could announce the post titles of all the fetched items? 

@alexstine, do you have guidance for what the preferred pattern is? I've looked around and couldn't find much.

Reply:
I think just the loading announcement is fine followed by focus on the first item.